Let/Let.md

```diff
@@ -0,0 +1,128 @@
+# Let-syntax for Proton
+
+###### Status: Draft
+###### Submitted: 2016-07-22
+###### Author: Soren Palmund / Miista
+
+---
+
+Introduction
+------------
+
+This proposal will introduce `let` expressions in two forms:
+
+1. As a way of introducing values into a local context
+2. As a way of declaring an immutable reference to some value
+
+1\) is referred to as the _expanded_ `let` expression or `let-in` expression,
+and 2) is referred to as a `let` declaration.
+
+Proposal
+--------
+
+This proposal will discuss the introduction of two forms of `let` expressions.
+Both forms allow declaration of variables with varying scope.
+
+For both the expanded `let` expression and the `let` declaration there is
+the possibility of allowing multiple declarations in the `let` block.
+I believe we should possibly start out with just allowing one declaration,
+as the later addition of multiple declarations is rather trivial.
+(On the paper and in the grammar, at least.)
+
+### Expanded `let`
+
+Usually one would declare and use a value as such:
+
+    double pi = 3.14;
+    double pi2 = pi * 2;
+
+After the declaration of `pi2` the value of `pi` lingers on and clutters
+the namespace when declaring a new variable.
+A `let` expression would be used to introduce scoping to a value declaration.
+The above example would be rewritten such that the variable `pi` only
+exist when declaring `pi2`.
+Using the suggested syntax for `let` expressions, the above example
+would be written as such:
+
+    double pi2 = let pi = 3.14 in pi * 2;
+
+Now, the variable `pi` exists only for the expression `pi * 2`.
+
+#### Mutability
+
+The expanded `let` expression allows declaration of a variable
+whose scope is only the `in` block of the `let` expression.
+The variable's value as set when it is initialized is immutable.
+That is, once declared it cannot be changed.
+
+#### Relation to Expression
+
+An expanded `let` expression would be equivalent to any
+other expression meaning that an expanded `let` expression could be used
+for a `return` statement in a function as such:
+
+    function Fun(b: Int) {
+      return let a = 2 in a + b
```

Comment:
The `return` here is merely there to indicate that we are returning from the function. However we end up doing it may be very different. This is just for demonstration purposes :)